src/Gaffer/ValuePlug.cpp

```diff
@@ -624,8 +626,24 @@ class ValuePlug::ComputeProcess : public Process
 			const ThreadState &threadState = ThreadState::current();
 			const Context *currentContext = threadState.context();
 
-			const ComputeNode *computeNode = IECore::runTimeCast<const ComputeNode>( p->node() );
-			const ComputeProcessKey processKey( p, plug, computeNode, computeNode ? computeNode->computeCachePolicy( p ) : CachePolicy::Uncached, precomputedHash );
+			CachePolicy cachePolicy = CachePolicy::Uncached;
+			if( computeNode )
+			{
+				if( !p->getInput() )
```

Comment:
I was concerned that adding the additional `getInput()` call here might affect the performance of `ValuePlugTest.testCacheOverhead()`, which is very sensitive to small additional overheads. So I measured, and we're actually coming out better by a few %. I assume this is because we're now only doing one `runTimeCast<ComputeNode>()` - nice one! I tried only doing one `getInput()` call and it made no further improvement.